### ISSUE TITLE
Tutorials/Instructions for using Refprop functions in Matlab

### ISSUE THREAD

**Issue by Yili-Zhang** [usnistgov/REFPROP-wrappers#94]
Hi All,

I wonder if anyone has clear and sufficient resource on how to use refprop (version 10)  functions in Matlab (2018)? I need some instructions to get started.

Any help would be very appreciated!

Thanks,

-Yili

**Comment by ianhbell** [comment]
Did you try the Tutorial I linked before: https://nbviewer.jupyter.org/github/usnistgov/REFPROP-wrappers/blob/master/wrappers/python/notebooks/Tutorial.ipynb ?  Its the same functions in MATLAB, since it is using the Python interface.

**Comment by Yili-Zhang** [comment]
Hi,

I tried a little bit, and the code broke in Matlab. Maybe I haven't understood enough, and would need a bit more time to look into it. I will get back to you after I investigated.

Thanks,

-Yili

**Comment by ianhbell** [comment]
If you can tell me what points are unclear in the tutorial, I'd appreciate that.  I want it to be useful to users of REFPROP!

**Comment by Yili-Zhang** [comment]
Hi,

Sorry for the delayed reply. I was fully occupied by other thing last week. 

After the RefProp was set up in MatLAb, I tried to run the code:
`MOLAR_BASE_SI = RP.GETENUMdll(0, "MOLAR BASE SI").iEnum`
to get the unit system, but I got the error:

> Error using ctREFPROP>GETENUMdll (line 1517)
Python Error: TypeError: int expected instead of float

If you may know what went wrong, please let me know and it would be very appreciated.

Thanks,

-Yili

**Comment by Yili-Zhang** [comment]
Hi,

After more reading and debugging, I think I figured out where the problem was. In the code, I changed the 0 into int8(0), and it worked. [https://refprop-docs.readthedocs.io/en/latest/DLL/high_level.html#f/_/REFPROPdll](url) This link is more helpful as its content is more basic.

The website you recommended is fine, but I found there are lots of minor differences between refprop in python and MatLAB, even through the frame work are pretty similar between them, but the minor difference sometimes may drive the new-users into nuts.

My suggestion as a user is to create a new set of instructions specific to MatLAB applications only, instead of borrowing the Python instructions. If it could happen, I believe more first-time users would benefit from it.

Thanks very much for all your help!

Best,

-Yili
